metals/src/main/scala/scala/meta/internal/metals/BuildTargetClasses.scala

```diff
@@ -87,8 +88,17 @@ final class BuildTargetClasses(
     for {
       item <- result.getItems.asScala
       target = item.getTarget
+      buildTarget <- buildTargets.scalaTarget(target)
       aClass <- item.getClasses.asScala
-      symbol <- createSymbols(aClass.getClassName, List(Descriptor.Term))
+      descriptors = {
+        if (ScalaVersions.isScala3Version(buildTarget.scalaVersion))
+          List(Descriptor.Term, Descriptor.Type)
```

Comment:
In case of Scala 3 we get additional reference for Descriptor.Type when using main methods, so we can use that. In the future it could change to a method, but that depends on the changes in Dotty.